util/src/main/scala/org/ensime/util/file.scala

```diff
@@ -115,6 +115,9 @@ package object file {
         case t: Throwable => file.getAbsoluteFile
       }
 
+    def descendsFrom(parent: File): Boolean =
+      file.getParent.startsWith(parent.getAbsolutePath)
```

Comment:
this needs a test. Also, there is a VFS approach to do this which may avoid the need to write the implementation.